### PR TITLE
added macOS Big Sur support

### PR DESCRIPTION
**Pull request by mpanighetti**
- added macOS Big Sur support
  - blocked Apple Silicon support until script can be validated and updated to support that platform architecture #45
  - added `--restart` flag to `softwareupdate` command for macOS Big Sur #47
- removed `%UPDATE_MECHANISM%` strings (mechanism will be consistent on all supported macOS versions in the near future)
- updated wording in message strings
- renamed plist attributes
- removed legacy resource file and package receipt references
- README formatting improvements
- removed CHANGELOG (GitHub releases serve this function better)

### CODE REVIEW

**Review by homebysix — COMMENTED** [review]
Did you mean to delete the change log?

File: README.md [hunk 1]
```diff
@@ -13,7 +13,8 @@ This framework is distributed in the form of a [munkipkg](https://github.com/mun
 
 Here's what needs to be in place in order to use this framework:
 
-- The current version of this framework has been tested only on __macOS 10.13 through 10.15__, but older versions should continue to function normally for previous macOS builds (note, however, that those versions of macOS are no longer receiving regular security updates from Apple and thus may not benefit from this framework).
+- The current version of this framework has been tested on __macOS High Sierra, Mojave, Catalina, and Big Sur__, but older versions should continue to function normally for previous macOS builds (note, however, that those versions of macOS are no longer receiving regular security updates from Apple and thus may not benefit from this framework).
+- This framework has only been tested on __Intel Macs__, and currently exits if run on Apple Silicon Macs. `softwareupdate` binary behavior has changed on Apple Silicon and further testing on native hardware is required before we can update the script for compatibility. Stay tuned!
```
Comment (homebysix) [inline README.md:17]: Possibly throw a link here to the GH issue so people can follow along?

File: payload/Library/Scripts/Install or Defer.sh [hunk 2]
```diff
@@ -12,8 +12,8 @@
 #                   the system restarts automatically.
 #         Authors:  Mario Panighetti and Elliot Jordan
 #         Created:  2017-03-09
-#   Last Modified:  2020-07-06
-#         Version:  3.0.2
+#   Last Modified:  2020-11-17
+#         Version:  3.0.3
```
Comment (homebysix) [inline payload/Library/Scripts/Install or Defer.sh:16]: I think a change as major as changing plist key names should be accompanied by a version bump from 3.0.2 to 4.0.0 (indicating [non-backwards-compatible](https://semver.org/)).

**Comment by mpanighetti** [comment]
Yeah, I was thinking that the notes on each GitHub release served the function sufficiently without manually copying everything to a separate changelog file.

**Comment by homebysix** [comment]
OK sounds good. There wasn't a separate commit, so just wanted to make sure it was intentional.

File: README.md (hunk 1 above)
Comment (mpanighetti) [reply README.md]: Added!

File: payload/Library/Scripts/Install or Defer.sh (hunk 2 above)
Comment (mpanighetti) [reply payload/Library/Scripts/Install or Defer.sh]: I was saving the major version change for eventually dropping High Sierra support, but that makes perfect sense. :) Bumped to 4.0.

**Review by homebysix — APPROVED** [review]

**Pull request merged**